src/plugins/discover/public/application/main/components/top_nav/get_top_nav_links.tsx

```diff
@@ -183,7 +183,9 @@ export const getTopNavLinks = ({
         objectId: savedSearch.id,
         objectType: 'search',
         sharingData: {
-          ...sharingData,
+          isTextBased,
+          locatorParams: [{ id: locator.id, params }],
```

Comment:
locatorParams will be used for es|ql report (csv_v2), searchSourceSharingData is still used for other reports. 
Hopefully we will only need locatorParams when fully migrated to csv_v2 https://github.com/elastic/kibana/issues/151190